tools/plugin-toolkit/src/_base/cli/prompts.ts

```diff
@@ -0,0 +1,140 @@
+// (C) 2007-2021 GoodData Corporation
+import { DistinctQuestion, prompt } from "inquirer";
+import { TargetAppFlavor, TargetBackendType } from "../types";
+
+export async function promptUsername(wording: string = "username"): Promise<string> {
+    const usernameQuestion: DistinctQuestion = {
+        type: "input",
+        name: "username",
+        message: `Enter your ${wording}:`,
+    };
+    const usernameResponse = await prompt(usernameQuestion);
+
+    return usernameResponse.username;
+}
+
+export async function promptPassword(): Promise<string> {
+    const passwordQuestion: DistinctQuestion = {
+        type: "password",
+        name: "password",
+        message: "Enter your password:",
+    };
+    const passwordResponse = await prompt(passwordQuestion);
+
+    return passwordResponse.password;
+}
+
+export type WorkspaceChoices = {
+    name: string;
+    value: string;
+};
+
+export async function promptWorkspaceId(choices: WorkspaceChoices[]): Promise<string> {
+    const question: DistinctQuestion = {
+        type: "list",
+        name: "id",
+        message: `Choose a workspace:`,
+        choices,
+    };
+
+    const response = await prompt(question);
+    return response.id;
+}
+
+export async function promptName(object: string = "dashboard plugin"): Promise<string> {
+    const question: DistinctQuestion = {
+        message: `What is your ${object} name?`,
+        name: "name",
+        type: "input",
+    };
+
+    const response = await prompt(question);
+    return response.name;
+}
+
+export async function promptHostname(isBear: boolean): Promise<string> {
+    if (isBear) {
+        const question: DistinctQuestion = {
+            message: "What is your hostname?",
+            name: "hostname",
+            type: "list",
+            choices: [
+                {
+                    value: "https://secure.gooddata.com",
+                },
+                {
+                    value: "https://developer.na.gooddata.com",
+                },
+                {
+                    value: "https://salesengineering.na.gooddata.com",
+                },
+                {
+                    name: "I have a custom hostname",
+                    value: "WHITE_LABELLED",
+                },
+            ],
+        };
+
+        const response = await prompt(question);
+
+        if (response.hostname !== "WHITE_LABELLED") {
+            return response.hostname;
+        }
+    }
+
+    const question: DistinctQuestion = {
+        message: "Insert your hostname. Protocol defaults to https if none is provided.",
```

Comment:
The message is a bit misleading considering that on tiger the defaulting will not happen, right?

Reply:
took this from boilerplate. will check it out